hkmc2/shared/src/main/scala/hkmc2/codegen/Lifter.scala

```diff
@@ -0,0 +1,1442 @@
+    * but is not defined *by* `d`.
+    *
+    * @param accessed Previously defined locals which could possibly be accessed or mutated.
+    * @param mutated Such locals which could also be mutated by this definition.
+    * @param refdDefns Previously defined definitions which could possibly be used by this definition.
+    */
+  case class AccessInfo(
+      accessed: Set[Local], 
+      mutated: Set[Local], 
+      refdDefns: Set[BlockMemberSymbol]
+    ):
+    def ++(that: AccessInfo) = AccessInfo(
+        accessed ++ that.accessed,
+        mutated ++ that.mutated,
+        refdDefns ++ that.refdDefns
+      )
+    def withoutLocals(locals: Set[Local]) = AccessInfo(
+        accessed -- locals,
+        mutated -- locals,
+        refdDefns
+      )
+    def intersectLocals(locals: Set[Local]) = AccessInfo(
+        accessed.intersect(locals),
+        mutated.intersect(locals),
+        refdDefns
+      )
+    def withoutBms(locals: Set[BlockMemberSymbol]) = AccessInfo(
+        accessed,
+        mutated,
+        refdDefns -- locals
+      )
+    def intersectBms(locals: Set[BlockMemberSymbol]) = AccessInfo(
+        accessed,
+        mutated,
+        refdDefns.intersect(locals)
+      )
+    def addAccess(l: Local) = copy(accessed = accessed + l)
+    def addMutated(l: Local) = copy(accessed = accessed + l, mutated = mutated + l)
+    def addRefdDefn(l: BlockMemberSymbol) = copy(refdDefns = refdDefns + l)
+    
+  object AccessInfo:
+    val empty = AccessInfo(Set.empty, Set.empty, Set.empty)
+
+  def getVars(d: Defn)(using state: State): Set[Local] = d match
+    case f: FunDefn =>
+      (f.body.definedVars ++ f.params.flatMap(_.paramSyms)).collect:
+        case s: FlowSymbol if !(s is state.runtimeSymbol) => s
+    case c: ClsLikeDefn =>
+      (c.preCtor.definedVars ++ c.ctor.definedVars).collect:
+        case s: FlowSymbol if !(s is state.runtimeSymbol) => s
+    case _ => Set.empty
+
+  def getVarsBlk(b: Block)(using state: State): Set[Local] =
+    b.definedVars.collect:
+      case s: FlowSymbol if !(s is state.runtimeSymbol) => s
+
+  object RefOfBms:
+    def unapply(p: Path) = p match
+      case Value.Ref(l: BlockMemberSymbol) => S(l)
+      case s @ Select(_, _) => s.symbol match
+        case Some(value: BlockMemberSymbol) => S(value)
+        case _ => N
+      case _ => N
+  
+  object InstSel:
+    def unapply(p: Path) = p match
+      case Value.Ref(l: BlockMemberSymbol) => S(l)
+      case s @ Select(Value.Ref(l: BlockMemberSymbol), Tree.Ident("class")) => S(l)
+      case _ => N
+
+  def modOrObj(d: Defn) = d match
+    case c: ClsLikeDefn => (c.k is syntax.Mod) || (c.k is syntax.Obj)
+    case _ => false
+
+  // TODO: this is very unclean
+  def isContClassPth(p: Path)(using s: State) = p match
+    case Select(Select(Select(Value.Ref(s.globalThisSymbol), Tree.Ident("Predef")), 
+      Tree.Ident("__Cont")), Tree.Ident("class")) => true
+    case _ => false
+
```

Comment:
Expose the path in HandlerLowering and access it here, the path already changed in the handler runtime PR so we won't forget to update the path

Reply:
You can also just merge the other PR into this one. The other PR will most likely be merged first.

Reply:
Actually it's merged now.